File: dmn-core/src/test/resources/tck/cl3/expected/dmn/0031-user-defined-functions/type/TFnInvocationComplexParamsResultImpl.java
package type;

import java.util.*;

@javax.annotation.Generated(value = {"itemDefinition.ftl", "tFnInvocationComplexParamsResult"})
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
public class TFnInvocationComplexParamsResultImpl implements TFnInvocationComplexParamsResult {
        private java.math.BigDecimal functionInvocationLiteralExpressionInParameter;
        private java.math.BigDecimal functionInvocationInParameter;
        private java.math.BigDecimal circumference;

    public TFnInvocationComplexParamsResultImpl() {
    }

    public TFnInvocationComplexParamsResultImpl(java.math.BigDecimal circumference, java.math.BigDecimal functionInvocationInParameter, java.math.BigDecimal functionInvocationLiteralExpressionInParameter) {
        this.setCircumference(circumference);
        this.setFunctionInvocationInParameter(functionInvocationInParameter);
        this.setFunctionInvocationLiteralExpressionInParameter(functionInvocationLiteralExpressionInParameter);
    }

    @com.fasterxml.jackson.annotation.JsonGetter("circumference")
    public java.math.BigDecimal getCircumference() {
        return this.circumference;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("circumference")
    public void setCircumference(java.math.BigDecimal circumference) {
        this.circumference = circumference;
    }

    @com.fasterxml.jackson.annotation.JsonGetter("functionInvocationInParameter")
    public java.math.BigDecimal getFunctionInvocationInParameter() {
        return this.functionInvocationInParameter;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("functionInvocationInParameter")
    public void setFunctionInvocationInParameter(java.math.BigDecimal functionInvocationInParameter) {
        this.functionInvocationInParameter = functionInvocationInParameter;
    }

    @com.fasterxml.jackson.annotation.JsonGetter("functionInvocationLiteralExpressionInParameter")
    public java.math.BigDecimal getFunctionInvocationLiteralExpressionInParameter() {
        return this.functionInvocationLiteralExpressionInParameter;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("functionInvocationLiteralExpressionInParameter")
    public void setFunctionInvocationLiteralExpressionInParameter(java.math.BigDecimal functionInvocationLiteralExpressionInParameter) {
        this.functionInvocationLiteralExpressionInParameter = functionInvocationLiteralExpressionInParameter;
    }

    @Override
    public boolean equals(Object o) {
        return equalTo(o);
    }

    @Override
    public int hashCode() {
        return hash();
    }

    @Override
    public String toString() {
        return asString();
    }
}

File: dmn-test-cases/tck/1.1/cl3/0031-user-defined-functions/translator/expected/kotlin/dmn/type/TFnInvocationComplexParamsResultImpl.kt
package type

import java.util.*

@javax.annotation.Generated(value = ["itemDefinition.ftl", "tFnInvocationComplexParamsResult"])
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
class TFnInvocationComplexParamsResultImpl : TFnInvocationComplexParamsResult {
    @get:com.fasterxml.jackson.annotation.JsonGetter("functionInvocationLiteralExpressionInParameter")
    @set:com.fasterxml.jackson.annotation.JsonGetter("functionInvocationLiteralExpressionInParameter")
    override var functionInvocationLiteralExpressionInParameter: java.math.BigDecimal? = null

    @get:com.fasterxml.jackson.annotation.JsonGetter("functionInvocationInParameter")
    @set:com.fasterxml.jackson.annotation.JsonGetter("functionInvocationInParameter")
    override var functionInvocationInParameter: java.math.BigDecimal? = null

    @get:com.fasterxml.jackson.annotation.JsonGetter("circumference")
    @set:com.fasterxml.jackson.annotation.JsonGetter("circumference")
    override var circumference: java.math.BigDecimal? = null

    constructor() {
    }

    constructor (circumference: java.math.BigDecimal?, functionInvocationInParameter: java.math.BigDecimal?, functionInvocationLiteralExpressionInParameter: java.math.BigDecimal?) {
        this.circumference = circumference
        this.functionInvocationInParameter = functionInvocationInParameter
        this.functionInvocationLiteralExpressionInParameter = functionInvocationLiteralExpressionInParameter
    }

    override fun equals(other: Any?): Boolean {
        return equalTo(other)
    }

    override fun hashCode(): Int {
        return hash()
    }

    @Override
    override fun toString(): String {
        return asString()
    }
}
